scripts/promote.py

```diff
@@ -181,57 +172,50 @@ def update_version() -> str:
     return new_version
 
 
-Release_msg = "Releasing {src} to {dst}"
-if args.dry_run:
-    Release_msg = Release_msg + " (dry run)"
-Release_name = "{dst} {new_version}"
-release_map = {
-    "integration": ("master", "integration", True),
-    "staging": ("integration", "staging", True),
-    "production": ("staging", "production", False)
-}
-
-s = requests.Session()
-token_path = os.environ.get('GITHUB_TOKEN_PATH')
-if token_path and token_path != 'None':
-    with open(os.path.expanduser(token_path), 'r') as fp:
-        token = fp.read().strip()
-else:
-    secret_id = os.environ['GITHUB_TOKEN_SECRET_NAME']
-    import boto3
-
-    SM = boto3.client('secretsmanager')
-    token = SM.get_secret_value(SecretId=secret_id)['SecretString']
-
 if __name__ == "__main__":
+    release_map = {
+        "integration": ("master", "integration", True),
+        "staging": ("integration", "staging", True),
+        "production": ("staging", "production", False)
+    }
+
+    token_path = os.environ.get('GITHUB_TOKEN_PATH')
+    if token_path and token_path != 'None':
+        with open(os.path.expanduser(token_path), 'r') as fp:
+            token = fp.read().strip()
+    else:
+        secret_id = os.environ['GITHUB_TOKEN_SECRET_NAME']
+        import boto3
+
+        SM = boto3.client('secretsmanager')
+        token = SM.get_secret_value(SecretId=secret_id)['SecretString']
+
     src, dst, prerelease = release_map[args.stage]
-    print(Release_msg.format(src=src, dst=dst))
+    dry_run = "(dry run)" if args.dry_run else ""
+    print(f"Releasing {src} to {dst} {dry_run}")
     print(_subprocess(['git', 'fetch', '--all']))
     check_working_tree()
     check_diff(src, dst)
     release_notes = make_release_notes(src, dst)
     new_version = update_version()
     if not args.dry_run:
         commit(src, dst)
-        name = Release_name.format(dst=dst, new_version=new_version)
         body = dict(
             tag_name=str(new_version),
-            name=name,
+            name="{dst} {new_version}".format(dst=dst, new_version=new_version),
             prerelease=prerelease,
             draft=False,
             target_commitish=dst,
             body=release_notes
         )
-        resp = s.post(
+
+        resp = requests.Session().post(
             f"https://api.github.com/repos/HumancellAtlas/fusillade/releases",
             headers={"Authorization": f"token {token}"},
             data=json.dumps(body)
         )
         try:
             resp.raise_for_status()
         except requests.exceptions.HTTPError as ex:
-            notes_path = os.path.abspath(f"release-notes-{name}.txt")
-            with open(notes_path, 'w') as fp:
-                fp.write(release_notes)
-            print(f"ERROR: Failed to create release!  Check log: {notes_path}")
+            print(f"ERROR: Failed to create release!  Changes were:\n{release_notes}")
```

Comment:
actually if this fails we should try to roll back the changes we made to git. Otherwise if we try to rerun this the branches will already be in sync and therefore there will be no changes in the release notes.